source/simulator/introspection.cc

```diff
@@ -170,11 +176,9 @@ namespace aspect
                              :
                              parameters.stokes_velocity_degree);
       quadratures.temperature = reference_cell.face_reference_cell(0).get_gauss_type_quadrature<dim-1>(parameters.temperature_degree+1);
-      quadratures.compositional_fields = reference_cell.face_reference_cell(0).get_gauss_type_quadrature<dim-1>(parameters.composition_degree+1);
-      quadratures.system = reference_cell.face_reference_cell(0).get_gauss_type_quadrature<dim-1>(std::max({parameters.stokes_velocity_degree,
-                           parameters.temperature_degree,
-                           parameters.composition_degree
-                                                                                                           }) + 1);
+      quadratures.compositional_fields = reference_cell.face_reference_cell(0).get_gauss_type_quadrature<dim-1>(parameters.max_composition_degree+1);
```

Comment:
So you use the highest compositional degree as the one that controls which quadrature to use when someone asks for the composition quadrature? Could this be a problem for performance, since it means that even for assembling lower order compositional fields we will use the maximum necessary quadrature order? Or would it make sense to store a vector of quadratures for the compositional fields and use a different one for each field?